OBJETIVOS:

Vendas Semanais
Comparar vendas semanais entre diferentes lojas (se houver mais de uma).
Analisar sazonalidade nas vendas (se houver).

Feriados 
Avaliar o impacto dos feriados nas vendas.
Comparar o desempenho de vendas em semanas com e sem feriados.
Identificar quais feriados têm maior impacto nas vendas.

Temperatura:
Investigar a relação entre a temperatura e as vendas.
Identificar padrões sazonais ou climáticos que afetam as vendas.

Preço do Combustível 
Analisar se há correlação entre o preço do combustível e as vendas.
Verificar se mudanças nos preços do combustível afetam o comportamento de compra dos clientes.
Analisar a relação entre a taxa de desemprego e as vendas.

Conclusões Finais

In [133]:
# Importando bibliotecas
import pandas as pd  
import matplotlib as plt  

In [134]:
base_walmart = pd.read_csv(r'Walmart_sales.csv')  # Lê o arquivo CSV e o armazena em um DataFrame chamado base_walmart
base_walmart.info()  # Exibe informações sobre o DataFrame

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         6435 non-null   int64  
 1   Date          6435 non-null   object 
 2   Weekly_Sales  6435 non-null   float64
 3   Holiday_Flag  6435 non-null   int64  
 4   Temperature   6435 non-null   float64
 5   Fuel_Price    6435 non-null   float64
 6   CPI           6435 non-null   float64
 7   Unemployment  6435 non-null   float64
dtypes: float64(5), int64(2), object(1)
memory usage: 402.3+ KB


In [135]:
base_walmart.describe().round(2)  # Gera estatísticas descritivas do DataFrame e arredonda para 2 casas decimais


Store  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price      CPI  \
count  6435.00       6435.00       6435.00      6435.00     6435.00  6435.00   
mean     23.00    1046964.88          0.07        60.66        3.36   171.58   
std      12.99     564366.62          0.26        18.44        0.46    39.36   
min       1.00     209986.25          0.00        -2.06        2.47   126.06   
25%      12.00     553350.10          0.00        47.46        2.93   131.74   
50%      23.00     960746.04          0.00        62.67        3.44   182.62   
75%      34.00    1420158.66          0.00        74.94        3.74   212.74   
max      45.00    3818686.45          1.00       100.14        4.47   227.23   

       Unemployment  
count       6435.00  
mean           8.00  
std            1.88  
min            3.88  
25%            6.89  
50%            7.87  
75%            8.62  
max           14.31

In [136]:
# Renomeia as colunas do DataFrame
base_walmart = base_walmart.rename(columns={
    'Store': 'LOJA',
    'Date': 'DATA',
    'Weekly_Sales': 'VENDA SEMANAL',
    'Holiday_Flag': 'FERIADO',
    'Temperature': 'TEMPERATURA',
    'Fuel_Price': 'PRECO COMBUSTIVEL',
    'Unemployment': 'TX MD SEMANAL DESEMPREGO',
    'CPI': 'VAR MED DE PRECOS POR FAMILIA'
})

base_walmart['DATA'] = pd.to_datetime(base_walmart['DATA'], format='%d-%m-%Y')  # Converte a coluna 'DATA' para o formato datetime


In [137]:
quantidade_lojas = base_walmart['LOJA'].unique()  # Obtém os valores únicos da coluna 'LOJA' 
print(quantidade_lojas)  # Exibe os valores únicos das lojas

aparicoes_loja = base_walmart['LOJA'].value_counts()  # Conta o número de ocorrências de cada loja e armazena em aparicoes_loja
#print(aparicoes_loja)  # Exibe o número de ocorrências de cada loja


[ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45]


In [142]:
display(base_walmart.sort_values(by='VENDA SEMANAL', ascending=False).head(20))

LOJA       DATA  VENDA SEMANAL  FERIADO  TEMPERATURA  PRECO COMBUSTIVEL  \
1905    14 2010-12-24     3818686.45        0        30.59              3.141   
2763    20 2010-12-24     3766687.43        0        25.17              3.141   
1333    10 2010-12-24     3749057.69        0        57.06              3.236   
527      4 2011-12-23     3676388.98        0        35.92              3.103   
1762    13 2010-12-24     3595903.20        0        34.90              2.846   
1814    13 2011-12-23     3556766.03        0        24.76              3.186   
2815    20 2011-12-23     3555371.03        0        40.19              3.389   
475      4 2010-12-24     3526713.39        0        43.21              2.887   
1385    10 2011-12-23     3487986.89        0        48.36              3.541   
189      2 2010-12-24     3436007.68        0        49.97              2.886   
1957    14 2011-12-23     3369068.99        0        42.27              3.389   
241      2 2011-12-23     3224369.80        0        46.66              3.112   
3764    27 2010-12-24     3078162.08        0        31.34              3.309   
523      4 2011-11-25     3004702.33        1        47.96              3.225   
1381    10 2011-11-25     2950198.64        1        60.68              3.760   
1329    10 2010-11-26     2939946.38        1        55.33              3.162   
1901    14 2010-11-26     2921709.71        1        46.15              3.039   
2811    20 2011-11-25     2906233.25        1        46.38              3.492   
1810    13 2011-11-25     2864170.61        1        38.89              3.445   
2762    20 2010-12-17     2819193.17        0        24.07              3.140   

      VAR MED DE PRECOS POR FAMILIA  TX MD SEMANAL DESEMPREGO  
1905                     182.544590                     8.724  
2763                     204.637673                     7.484  
1333                     126.983581                     9.003  
527                      129.984548                     5.143  
1762                     126.983581                     7.795  
1814                     129.984548                     6.392  
2815                     212.236040                     7.082  
475                      126.983581                     7.127  
1385                     129.984548                     7.874  
189                      211.064660                     8.163  
1957                     188.929975                     8.523  
241                      218.999550                     7.441  
3764                     136.597273                     8.021  
523                      129.836400                     5.143  
1381                     129.836400                     7.874  
1329                     126.669267                     9.003  
1901                     182.783277                     8.724  
2811                     211.412076                     7.082  
1810                     129.836400                     6.392  
2762                     204.632119                     7.484

Vendas Semanais:
  Comparar vendas semanais entre diferentes lojas (se houver mais de uma).
  Analisar sazonalidade nas vendas (se houver).

In [138]:
top_5_lojas = base_walmart.groupby('LOJA')['VENDA SEMANAL'].mean().sort_values(ascending=False).round(2)  # Calcula a média de vendas semanais para as lojas e ordena do maior para o menor
print(top_5_lojas.head(5), '\n')  # Exibe as 5 lojas com maiores médias de vendas semanais

top_5_less = base_walmart.groupby('LOJA')['VENDA SEMANAL'].mean().sort_values(ascending=True).round(2)  # Calcula a média de vendas semanais para as lojas e ordena do menor para o maior
print(top_5_less.head(5), '\n')  # Exibe as 5 lojas com menores médias de vendas semanais

desvio_padrao = top_5_lojas.std().round(2)  # Calcula o desvio padrão das médias de vendas semanais das lojas
print(f'Desvio Padrão de vendas semanais: {desvio_padrao}')  # Exibe o desvio padrão das médias de vendas semanais

l1 = base_walmart.loc[base_walmart['LOJA'] == 20]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas semanais para a loja 20
l2 = base_walmart.loc[base_walmart['LOJA'] == 4]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas semanais para a loja 4
l3 = base_walmart.loc[base_walmart['LOJA'] == 14]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas semanais para a loja 14
l4 = base_walmart.loc[base_walmart['LOJA'] == 13]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas semanais para a loja 13
l5 = base_walmart.loc[base_walmart['LOJA'] == 2]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas semanais para a loja 2

diferenca_1_2 = l1 - l2  # Calcula a diferença entre as médias de vendas semanais das lojas 1 e 2
diferenca_2_3 = l2 - l3  # Calcula a diferença entre as médias de vendas semanais das lojas 2 e 3
diferenca_3_4 = l3 - l4  # Calcula a diferença entre as médias de vendas semanais das lojas 3 e 4
diferenca_4_5 = l4 - l5  # Calcula a diferença entre as médias de vendas semanais das lojas 4 e 5

media_diferencas = ((diferenca_1_2 + diferenca_2_3 + diferenca_3_4 + diferenca_4_5) / 4).round(2)  # Calcula a média das diferenças entre as 5 lojas com maiores médias de vendas semanais
print(f'Média das diferenças entre as 5 lojas que mais rentabilizaram: {media_diferencas}')  # Exibe a média das diferenças

LOJA
20    2107676.87
4     2094712.96
14    2020978.40
13    2003620.31
2     1925751.34
Name: VENDA SEMANAL, dtype: float64 

LOJA
33    259861.69
44    302748.87
5     318011.81
36    373511.99
38    385731.65
Name: VENDA SEMANAL, dtype: float64 

Desvio Padrão de vendas semanais: 546626.27
Média das diferenças entre as 5 lojas que mais rentabilizaram: 45481.38


Feriados (Holiday_Flag):
  Avaliar o impacto dos feriados nas vendas.
  Comparar o desempenho de vendas em semanas com e sem feriados.
  Identificar quais feriados têm maior impacto nas vendas.

In [139]:

correlacao_feriado_venda = base_walmart['FERIADO'].corr(base_walmart['VENDA SEMANAL'])  # Calcula a correlação entre os dias de feriado e as vendas semanais
print(f'Correlação entre dias de feriado e vendas da semana: {correlacao_feriado_venda:.3f}')  # Exibe a correlação entre feriados e vendas semanais

lojas_desejadas_top = [20, 4, 14, 13, 2]  # Lista das lojas desejadas (com maiores médias de vendas)

df_top5 = base_walmart[base_walmart['LOJA'].isin(lojas_desejadas_top)]  # Seleciona as linhas das 5 lojas desejadas
df_top5 = df_top5.sort_values(by='VENDA SEMANAL', ascending=False)  # Ordena as linhas pelo valor das vendas semanais de forma decrescente
display(df_top5.head(20))  # Exibe as primeiras 20 linhas do DataFrame resultante

lojas_desejadas_lower = [33, 44, 5, 36, 8]  # Lista das lojas desejadas (com menores médias de vendas)

df_lower5 = base_walmart[base_walmart['LOJA'].isin(lojas_desejadas_lower)]  # Seleciona as linhas das 5 lojas desejadas
display(df_lower5.head(20))  # Exibe as primeiras 20 linhas do DataFrame resultante

media_sem_feriados = base_walmart[base_walmart['FERIADO'] == 1]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas em dias de feriado
print(f'Média de vendas em dias que são feriados: {media_sem_feriados}')  # Exibe a média de vendas em dias de feriado

media_com_feriados = base_walmart[base_walmart['FERIADO'] == 0]['VENDA SEMANAL'].mean().round(2)  # Calcula a média de vendas em dias sem feriado
print(f'Média de vendas em dias que não são feriados: {media_com_feriados}')  # Exibe a média de vendas em dias sem feriado


Correlação entre dias de feriado e vendas da semana: 0.037


LOJA       DATA  VENDA SEMANAL  FERIADO  TEMPERATURA  PRECO COMBUSTIVEL  \
1905    14 2010-12-24     3818686.45        0        30.59              3.141   
2763    20 2010-12-24     3766687.43        0        25.17              3.141   
527      4 2011-12-23     3676388.98        0        35.92              3.103   
1762    13 2010-12-24     3595903.20        0        34.90              2.846   
1814    13 2011-12-23     3556766.03        0        24.76              3.186   
2815    20 2011-12-23     3555371.03        0        40.19              3.389   
475      4 2010-12-24     3526713.39        0        43.21              2.887   
189      2 2010-12-24     3436007.68        0        49.97              2.886   
1957    14 2011-12-23     3369068.99        0        42.27              3.389   
241      2 2011-12-23     3224369.80        0        46.66              3.112   
523      4 2011-11-25     3004702.33        1        47.96              3.225   
1901    14 2010-11-26     2921709.71        1        46.15              3.039   
2811    20 2011-11-25     2906233.25        1        46.38              3.492   
1810    13 2011-11-25     2864170.61        1        38.89              3.445   
2762    20 2010-12-17     2819193.17        0        24.07              3.140   
2759    20 2010-11-26     2811634.04        1        46.66              3.039   
471      4 2010-11-26     2789469.45        1        48.08              2.752   
1761    13 2010-12-17     2771646.81        0        35.21              2.842   
526      4 2011-12-16     2771397.17        0        36.44              3.149   
1758    13 2010-11-26     2766400.05        1        28.22              2.830   

      VAR MED DE PRECOS POR FAMILIA  TX MD SEMANAL DESEMPREGO  
1905                     182.544590                     8.724  
2763                     204.637673                     7.484  
527                      129.984548                     5.143  
1762                     126.983581                     7.795  
1814                     129.984548                     6.392  
2815                     212.236040                     7.082  
475                      126.983581                     7.127  
189                      211.064660                     8.163  
1957                     188.929975                     8.523  
241                      218.999550                     7.441  
523                      129.836400                     5.143  
1901                     182.783277                     8.724  
2811                     211.412076                     7.082  
1810                     129.836400                     6.392  
2762                     204.632119                     7.484  
2759                     204.962100                     7.484  
471                      126.669267                     7.127  
1761                     126.879484                     7.795  
526                      129.898065                     5.143  
1758                     126.669267                     7.795

LOJA       DATA  VENDA SEMANAL  FERIADO  TEMPERATURA  PRECO COMBUSTIVEL  \
572     5 2010-02-05      317173.10        0        39.70              2.572   
573     5 2010-02-12      311825.70        1        39.81              2.548   
574     5 2010-02-19      303447.57        0        41.14              2.514   
575     5 2010-02-26      270281.63        0        46.70              2.561   
576     5 2010-03-05      288855.71        0        48.89              2.625   
577     5 2010-03-12      297293.59        0        58.50              2.667   
578     5 2010-03-19      281706.41        0        55.46              2.720   
579     5 2010-03-26      273282.97        0        52.47              2.732   
580     5 2010-04-02      331406.00        0        63.18              2.719   
581     5 2010-04-09      328020.49        0        65.19              2.770   
582     5 2010-04-16      306858.69        0        65.30              2.808   
583     5 2010-04-23      288839.73        0        65.13              2.795   
584     5 2010-04-30      298697.84        0        67.53              2.780   
585     5 2010-05-07      333522.60        0        71.53              2.835   
586     5 2010-05-14      296673.77        0        74.79              2.854   
587     5 2010-05-21      301615.49        0        75.20              2.826   
588     5 2010-05-28      310013.11        0        79.81              2.759   
589     5 2010-06-04      337825.89        0        79.54              2.705   
590     5 2010-06-11      296641.91        0        80.70              2.668   
591     5 2010-06-18      313795.60        0        83.91              2.637   

     VAR MED DE PRECOS POR FAMILIA  TX MD SEMANAL DESEMPREGO  
572                     211.653972                     6.566  
573                     211.800470                     6.566  
574                     211.847128                     6.566  
575                     211.877147                     6.566  
576                     211.907165                     6.566  
577                     211.937184                     6.566  
578                     211.770897                     6.566  
579                     211.571892                     6.566  
580                     211.372888                     6.465  
581                     211.173883                     6.465  
582                     211.038853                     6.465  
583                     210.989120                     6.465  
584                     210.939388                     6.465  
585                     210.889656                     6.465  
586                     210.887277                     6.465  
587                     211.169023                     6.465  
588                     211.450769                     6.465  
589                     211.732515                     6.465  
590                     212.014261                     6.465  
591                     212.011977                     6.465

Média de vendas em dias que são feriados: 1122887.89
Média de vendas em dias que não são feriados: 1041256.38


Temperatura:
  Investigar a relação entre a temperatura e as vendas.
  Identificar padrões sazonais ou climáticos que afetam as vendas.

In [140]:
correlacao_temperatura_vendas = base_walmart['TEMPERATURA'].corr(base_walmart['VENDA SEMANAL'])  # Calcula a correlação entre temperatura e vendas
print(f'Correlação entre temperatura e valor de vendas: {correlacao_temperatura_vendas:.5f}')  # Exibe a correlação entre temperatura e vendas

media_temperatura_geral = base_walmart['TEMPERATURA'].mean()  # Calcula a média da temperatura geral

top_20_vendas = base_walmart.sort_values(by='VENDA SEMANAL', ascending=False).head(20)  # Seleciona as 20 maiores vendas
lower_20_vendas = base_walmart.sort_values(by='VENDA SEMANAL', ascending=True).head(20)  # Seleciona as 20 menores vendas

media_temperatura_TOP20 = top_20_vendas['TEMPERATURA'].mean()  # Calcula a média da temperatura das 20 maiores vendas
media_temperatura_LOWER20 = lower_20_vendas['TEMPERATURA'].mean()  # Calcula a média da temperatura das 20 menores vendas

TempCelsius_geral = ((media_temperatura_geral - 32) / 1.8)  # Converte a média de temperatura geral para Celsius
TempCelsius_top = ((media_temperatura_TOP20 - 32) / 1.8)  # Converte a média de temperatura das 20 maiores vendas para Celsius
TempCelsius_lower = ((media_temperatura_LOWER20 - 32) / 1.8)  # Converte a média de temperatura das 20 menores vendas para Celsius

print(f'Média temperatura total: {TempCelsius_geral:.2f}°C\nMédia temperatura top 20 vendas: {TempCelsius_top:.2f}°C\nMédia temperatura piores 20 vendas: {TempCelsius_lower:.2f}°C')  # Exibe as médias de temperatura


Correlação entre temperatura e valor de vendas: -0.06381
Média temperatura total: 15.92°C
Média temperatura top 20 vendas: 5.27°C
Média temperatura piores 20 vendas: 24.01°C


Preço do Combustível (Fuel_Price):
  Analisar se há correlação entre o preço do combustível e as vendas.
  Verificar se mudanças nos preços do combustível afetam o comportamento de compra dos clientes.
  Analisar a relação entre a taxa de desemprego e as vendas.

In [141]:
media_valor_combustivel = base_walmart['PRECO COMBUSTIVEL'].mean().round(2)  # Calcula a média do preço do combustível

top_10_vendas = base_walmart.sort_values(by='VENDA SEMANAL', ascending=False).head(10)  # Seleciona as 10 maiores vendas
media_top_10 = top_10_vendas['PRECO COMBUSTIVEL'].mean().round(2)  # Calcula a média do preço do combustível das 10 maiores vendas

low_10_vendas = base_walmart.sort_values(by='VENDA SEMANAL', ascending=True).head(10)  # Seleciona as 10 menores vendas
media_lower_10 = low_10_vendas['PRECO COMBUSTIVEL'].mean().round(2)  # Calcula a média do preço do combustível das 10 menores vendas

print(f'Média geral de combustível: {media_valor_combustivel} \nMédia combustível top 10 vendas: {media_top_10} \nMédia combustível 10 piores vendas: {media_lower_10}')  # Exibe as médias do preço do combustível

correlacao_combust_familia = base_walmart['PRECO COMBUSTIVEL'].corr(base_walmart['VAR MED DE PRECOS POR FAMILIA'])  # Calcula a correlação entre o preço do combustível e a média de consumo por família
print(f'Correlação entre preço do combustível e média de consumo por família: {correlacao_combust_familia:.5f}')  # Exibe a correlação entre preço do combustível e média de consumo por família

correlacao_desemprego_vendas = base_walmart['TX MD SEMANAL DESEMPREGO'].corr(base_walmart['VENDA SEMANAL'])  # Calcula a correlação entre a taxa de desemprego e as vendas semanais
print(f'Correlação entre tx de desemprego e venda semanal: {correlacao_desemprego_vendas:.5f}')  # Exibe a correlação entre a taxa de desemprego e as vendas semanais

TX_desemp_lower_10 = low_10_vendas['TX MD SEMANAL DESEMPREGO'].mean().round(2)  # Calcula a média da taxa de desemprego das 10 menores vendas
TX_desemp_top_10 = top_10_vendas['TX MD SEMANAL DESEMPREGO'].mean().round(2)  # Calcula a média da taxa de desemprego das 10 maiores vendas

print(f'Taxa média de desemprego top 10 semanas com mais vendas: {TX_desemp_top_10}\nTaxa média desemprego 10 piores semanas de venda: {TX_desemp_lower_10}')  # Exibe as médias da taxa de desemprego


Média geral de combustível: 3.36 
Média combustível top 10 vendas: 3.14 
Média combustível 10 piores vendas: 3.31
Correlação entre preço do combustível e média de consumo por família: -0.17064
Correlação entre tx de desemprego e venda semanal: -0.10618
Taxa média de desemprego top 10 semanas com mais vendas: 7.48
Taxa média desemprego 10 piores semanas de venda: 8.87


Com base nas análises realizadas:

1. **Vendas Semanais:**
   - As lojas mais destacadas em termos de vendas semanais são as de números 20, 4, 14, 13 e 2, enquanto as menos destacadas são as de números 33, 44, 5, 36 e 38.
   - Sem informações geográficas específicas, é difícil determinar os fatores sociológicos por trás das variações nos valores médios de venda entre as lojas.

2. **Feriados:**
   - A correlação entre dias de feriado e vendas semanais é insignificante (0.037).
   - Embora não haja uma correlação direta entre feriados e aumento nas vendas, é interessante notar que as maiores arrecadações ocorreram em semanas comemorativas como Dia de Ação de Graças, Natal e Black Friday.
   - A sazonalidade de vendas é mais pronunciada no final do ano, especialmente em 2010 e 2011.
   - A média de vendas em dias de feriado é consideravelmente maior do que em dias normais.

3. **Temperatura:**
   - A correlação entre temperatura e vendas é insignificante (-0.06381).
   - Apesar disso, observa-se que as maiores vendas semanais ocorreram nos meses finais do ano.
   - As médias de temperatura para as 20 melhores e 20 piores semanas de vendas variam consideravelmente.

4. **Combustível e Taxa de Desemprego:**
   - Existe uma correlação fraca entre o preço do combustível e a média de consumo por família (-0.17064), indicando uma tendência inversa: quanto mais alto o preço do combustível, menor o consumo.
   - Similarmente, há uma correlação inversa entre a taxa de desemprego e as vendas semanais (-0.10618), sugerindo que um aumento na taxa de desemprego está associado a uma diminuição nas vendas.
   - As médias de preço do combustível e taxa de desemprego para as 10 melhores e 10 piores semanas de vendas mostram variações consistentes com essas correlações.

Em resumo, enquanto algumas tendências são observadas, como o aumento nas vendas durante feriados e a relação inversa entre preço do combustível/taxa de desemprego e vendas, outras variáveis, como temperatura, não parecem ter uma influência significativa nas vendas semanais.